### PR TITLE
[exporter/datadog] Correctly derive status for logs from 'level'

### PR DESCRIPTION
**Pull request by dineshg13**
**Description:**  Bug fix for recently added logs support in Datadog
When a log has `level` field, we want to use that for status rather than try to derive if from `severityNumber` .  

**Link to tracking Issue:**  https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/2651

**Testing:** Tested by deploying demo application.

**Documentation:** Documentation will be added in followup PR

### CODE REVIEW

File: exporter/datadogexporter/internal/logs/translator.go [hunk 1]
```diff
@@ -75,10 +75,14 @@ func Transform(lr plog.LogRecord, res pcommon.Resource) datadogV2.HTTPLogItem {
...
 	// we need to set log attributes as AdditionalProperties
 	// AdditionalProperties are treated as Datadog Log Attributes
+	var status string
 	lr.Attributes().Range(func(k string, v pcommon.Value) bool {
 		switch strings.ToLower(k) {
-		case "msg", "message":
+		// set of remapping are taken from Datadog Backend
+		case "msg", "message", "log":
 			l.Message = v.AsString()
+		case "status", "severity", "level", "syslog.severity":
```
Comment (mx-psi) [inline exporter/datadogexporter/internal/logs/translator.go:84]: Where do these values come from?

**Review by mx-psi — APPROVED** [review]
Code LGTM, but please update the PR title and description to something more descriptive (what bug are you fixing in particular?)

File: exporter/datadogexporter/internal/logs/translator.go (hunk 1 above)
Comment (mx-psi) [reply exporter/datadogexporter/internal/logs/translator.go:84]: We discussed this offline, this comes from Datadog's backend code so we can't really link it here

**Comment by dineshg13** [comment]
updated the description

**Pull request merged**